source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -700,6 +860,108 @@ namespace
         }
     }
 
+    void writeCollections(PvtStage* stage, DocumentPtr dest, RtWriteOptions::WriteFilter filter)
```

Comment:
Should this be DocumentPtr& to avoid the increment?

Reply:
The C++ guidelines even go further: use a Document& in context where a function does not modify ownership.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-smartptrparam

Reply:
That makes sense actually.